### DISCUSSION TITLE
Databases, MGKDB and IMAS

### DISCUSSION THREAD

**Discussion by DanHarJor** [bpatel2107/pyrokinetics#385] [Q&A]
Hello, love the idea. Does pyrokinetic support or aim to support integration with gyrokinetic databases like MGKDB, https://github.com/Sapientai/MGKDB.

Does anyone know if it is the goal of IMAS to ensure that different gyrokinetic codes have the same input parameters file. If so then would it not make sense to take the prokinetics as the default for pyIMAS?

**Comment by bpatel2107** [comment]
We do have some people working on using pyrokinetic with MGKDB, @venkiteshayyar can comment on this better.

I don't think the different codes will have the same input file for variety of reasons, though it is possible that that all use say the `core_profiles` and `equilibrium` IDS files as input which I believe GENE is already working towards. This requires a lot of development and would require significant collaboration between different institutions which may take a long time.

We are of course happy to develop pyro to more smoothly integrate with any database, be it SimDB, GKDP or MGKDB. The only constraint I think should be that inputs/outputs should all be stored in a standard format (IMAS) to allow these databases to be combined at some point.

**Comment by venkiteshayyar** [comment]
@DanHarJor 
The goal of MGKDB is to enable code reproducibility, benchmarking, analysis etc. Hence, it is designed to store input and output files and the IMAS. We are currently using pyrokinetics to compute the IMAS for GENE, CGYRO, TGLF and GS2 and it's working out very well for us!
